### PR TITLE
Cockpit 200

### PR DESCRIPTION
**Pull request by martinpitt**
Rendered preview: https://martinpitt.github.io/cockpit-project.github.io/blog/cockpit-200.html

### CODE REVIEW

**Comment by martinpitt** [comment]
Release failed due to a test failure on s390x, we may need to respin for that. https://github.com/cockpit-project/cockpit/pull/12266 landed after that, so we need to include that.

File: _posts/2019-08-07-cockpit-200.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+category: release
+summary: Cockpit with new TLS proxy, firewall, and Machines improvements
+comments: 'true'
+---
+
+Cockpit is the [modern Linux admin interface](https://cockpit-project.org/). We release regularly. This is version Two Hundred, prepared for you live and in person from the Cockpit team sprint in Berlin!
+
+### Machines: Type-ahead OS selection
+
+The VM creation dialog merged the OS vendor and product inputs into a shared "Operating System" selector. This is more keyboard user friendly and supports type-ahead searching.
```
Comment (garrett) [inline _posts/2019-08-07-cockpit-200.md]:
```suggestion
In the VM creation dialog, OS vendor and product dropdowns have been replaced by a singular "Operating System" selector. The new dropdown is keyboard-friendly and supports auto-completion.
```

File: _posts/2019-08-07-cockpit-200.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+![Machines typeahead OS selection](/images/machines-os-typeahead.png)
+
+### Machines: Create LVM storage pools
+
+You can now create new VM storage pools on LVM:
```
Comment (garrett) [inline _posts/2019-08-07-cockpit-200.md]:
```suggestion
Storage pools using LVM can be created on the Machines page.
```

File: _posts/2019-08-07-cockpit-200.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+Cockpit 200 is available now:
+
+ * [For your Linux system](https://cockpit-project.org/running.html)
+ * [Source Tarball](https://github.com/cockpit-project/cockpit/releases/tag/200)
+ * [Fedora 30](https://bodhi.fedoraproject.org/updates/FEDORA-2019-d73df27df4)
+
+*[VM]: Virtual Machine
+*[LVM]: Logical Volume Manager
+*[OS]: Operating System
```
Comment (garrett) [inline _posts/2019-08-07-cockpit-200.md]:
```suggestion
*[OS]: Operating System
*[TLS]: Transport Layer Security
*[HTTP]: Hypertext Transport Protocol
```

**Comment by KKoukiou** [comment]
@martinpitt Just merged the Virtual Network Creation PR.
Let's add an entry like:

Machines: Create Virtual Networks
You can now create Virtual Networks to be used from NICs of VMs.
You can also attach the link to the libvirt documentation https://wiki.libvirt.org/page/Networking

File: _posts/2019-08-07-cockpit-200.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+### Networking: Show included firewalld services
+
+[firewalld 0.7.1](https://firewalld.org/2019/07/firewalld-0-7-1-release) introduces the concept of services that include other services. Cockpit's Firewall page shows these:
+
+![Firewall included services](/images/firewall-included-services.png)
+
+### Web server: Split out TLS handling
+
+This release introduces [cockpit-tls](https://cockpit-project.org/guide/latest/cockpit-tls.8.html), a TLS terminating HTTP proxy that runs in front of [cockpit-ws](https://cockpit-project.org/guide/latest/cockpit-ws.8.html). This allows us to move TLS termination out of cockpit-ws, as that is hard to audit and trust: That code is subject to external attacks through HTTP, does a lot of interpretation of data streams, uses a lot of external dependencies, and multiplexes sessions of different users in one process. Thus any vulnerability can easily lead to complete privilege escalation across all running and future sessions.
```
Comment (garrett) [inline _posts/2019-08-07-cockpit-200.md]:
```suggestion
This release introduces [`cockpit-tls`](https://cockpit-project.org/guide/latest/cockpit-tls.8.html), a TLS terminating HTTP proxy that runs in front of [cockpit-ws](https://cockpit-project.org/guide/latest/cockpit-ws.8.html).

Moving TLS termination out of `cockpit-ws` improves auditing and trust. The `cockpit-tls` proxy is subject to external attacks through HTTP, does a lot of interpretation of data streams, uses a lot of external dependencies, and multiplexes sessions of different users in one process.

This split provides a foundation for isolating user sessions in the future, to increase security hardening. It also forms the basis of future authentication methods, including the use of smart cards.
```

File: _posts/2019-08-07-cockpit-200.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+Right now this does not have any extra features over using cockpit-ws directly (other than better isolation), but it provides a basis for future work like supporting smart card authentication.
```
Comment (garrett) [inline _posts/2019-08-07-cockpit-200.md]:
```suggestion
```

File: _posts/2019-08-07-cockpit-200.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+This change is expected to be transparent to users. Please report issues if you encounter any page loading failures, stalls, or similar networking issues with this version!
```
Comment (garrett) [inline _posts/2019-08-07-cockpit-200.md]:
```suggestion
This change is expected to be transparent to users. _Please [report issues](https://github.com/cockpit-project/cockpit/issues/new?template=bug_report.md) if you encounter any page loading failures, stalls, or similar networking issues with this version!_
```

**Comment by martinpitt** [comment]
I added the new "create vm" feature and applied most of @garrett's suggestions. Let's clean up the cockpit-tls text tomorrow.

File: _posts/2019-08-07-cockpit-200.md [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+
+![Firewall included services](/images/firewall-included-services.png)
+
+### Web server: Split out TLS handling
+
+This release introduces [`cockpit-tls`](https://cockpit-project.org/guide/latest/cockpit-tls.8.html), a TLS terminating HTTP proxy that runs in front of [cockpit-ws](https://cockpit-project.org/guide/latest/cockpit-ws.8.html).
+
+Right now this does not have any extra features over using cockpit-ws directly (other than better isolation), but it provides a basis for future work like supporting smart card authentication.
+
+This change is expected to be transparent to users. Please [report issues](https://github.com/cockpit-project/cockpit/issues/new?template=bug_report.md) if you encounter any page loading failures, stalls, or similar networking issues with this version!
```
Comment (garrett) [inline _posts/2019-08-07-cockpit-200.md]:
```suggestion
This change is expected to be transparent to users. Please [report issues](https://github.com/cockpit-project/cockpit/issues/new?template=bug_report.md) if you encounter any page loading failures, stalls, or similar networking issues!
```

Comment (garrett) [reply _posts/2019-08-07-cockpit-200.md]: "with this version" is redundant

File: _posts/2019-08-07-cockpit-200.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+
+[firewalld 0.7.1](https://firewalld.org/2019/07/firewalld-0-7-1-release) introduces the concept of services that include other services. Cockpit's Firewall page shows these:
+
+![Firewall included services](/images/firewall-included-services.png)
+
+### Web server: Split out TLS handling
+
+This release introduces [`cockpit-tls`](https://cockpit-project.org/guide/latest/cockpit-tls.8.html), a TLS terminating HTTP proxy that runs in front of [cockpit-ws](https://cockpit-project.org/guide/latest/cockpit-ws.8.html).
+
+Right now this does not have any extra features over using cockpit-ws directly (other than better isolation), but it provides a basis for future work like supporting smart card authentication.
```
Comment (garrett) [inline _posts/2019-08-07-cockpit-200.md]:
```suggestion
At the moment, `cockpit-tls` does not provide additional features over using `cockpit-ws` directly (other than better isolation), but it sets up a basis for future work, such as supporting smart card authentication.
```

**Pull request merged**